frontend/app/DuplicateComponent.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import {RouteComponentProps, withRouter} from "react-router-dom";
+import { authenticatedFetch } from "./auth";
+import VaultSelector from "./searchnbrowse/NewVaultSelector";
+
+interface DuplicateComponentState {
+  vaultId?: string;
+  duplicatesCount?: string;
+  itemCount?: string;
+  duplicates?: DuplicateEntry[];
+}
+
+interface DuplicateEntry {
+  mxfsPath: string;
+  duplicateNumber: number;
+  duplicatesData: DuplicateData[];
+}
+
+interface DuplicateData {
+  mxfsPath: string;
+  oid: string;
+}
+
+class DuplicateComponent extends React.Component<RouteComponentProps, DuplicateComponentState> {
+
+  constructor(props:RouteComponentProps) {
+    super(props);
+
+    this.state = {
+      vaultId: undefined,
+      duplicatesCount: undefined,
+      itemCount: undefined,
+      duplicates: undefined
+    };
+  }
+
+  async getDuplicateData() {
+    const abortController = new AbortController();
+
+    const response = await authenticatedFetch('/api/vault/'+ this.state.vaultId +'/findDuplicates', {
+      signal: abortController.signal,
+    });
+    if (response.status !== 200) {
+      console.error(`Could not load data: server error ${response.status}`);
+      const rawData = await response.text();
+      console.error(`Server said ${rawData}`);
+
+      return;
+    } else {
+      const content = await response.json();
+      this.setState({ duplicatesCount: content.dupes_count });
```

Comment:
in a class-based container, you can set all three of these vars in the same call:

```jsx
this.setState({ duplicatesCount: ..., itemCount: ...., duplicates: ....});
```

Although TBH these days I would normally used a function-based component personally, as it's less typing :)

Reply:
Done.